core/include/detray/materials/detail/relativistic_quantities.hpp

```diff
@@ -151,6 +234,19 @@ struct relativistic_quantities {
         assert(m_betaGamma != 0.f);
         return -1.f * m_beta / (m_qOverP * (1.f + m_betaGamma * m_betaGamma));
     }
+
+    /// Compute derivative of betagamma w.r.t. q/p
+    ///
+    /// d(betagamma)/dqop = d(beta)/dqop * gamma + d(gamma)/dqop * beta
+    ///
+    /// Since d(gamma)/dqop = d(beta)/dqop * gamma * (beta / 1 - beta^2)
+    ///
+    /// d(betagamma)/dqop = gamma^3 * dbeta/dqop
+    DETRAY_HOST_DEVICE inline constexpr scalar_type derive_betaGamma() const {
+        assert(m_qOverP != 0.f);
```

Comment:
Why are these asserts here? Are they not in ```derive_beta``` already?

Reply:
Right. Fixed now